lib/AcqUnits/AcqUnitsField/AcqUnitsFieldContainer.js

```diff
@@ -0,0 +1,99 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { get, uniq } from 'lodash';
+
+import {
+  stripesConnect,
+  stripesShape,
+} from '@folio/stripes/core';
+
+import { LIMIT_MAX } from '../../constants';
+import { baseManifest } from '../../manifests';
+import { usePrevious } from '../../utils';
+
+import AcqUnitsField from './AcqUnitsField';
+
+export const buildAcqUnitsQuery = (units, isEdit) => {
+  const unitIds = uniq(units)
+    .map(acquisitionsUnitId => `id==${acquisitionsUnitId}`)
+    .join(' or ');
+
+  return isEdit ? unitIds : `${unitIds} or protectCreate==false`;
+};
+
+const AcqUnitsFieldContainer = ({
+  name, resources, mutator, stripes, preselectedUnits, perm, isEdit,
+}) => {
+  const userId = get(stripes, 'user.user.id');
+  const prevUserId = usePrevious(userId);
+
+  useEffect(() => {
+    if (userId !== prevUserId) {
+      mutator.acqUnitMemberships.reset();
+      mutator.acqUnits.reset();
+
+      mutator.acqUnitMemberships.GET({
+        params: {
+          limit: LIMIT_MAX,
+          query: `userId==${userId}`,
+        },
+      }).then(memberships => {
+        const units = [
+          memberships.map(({ acquisitionsUnitId }) => acquisitionsUnitId),
+          ...preselectedUnits,
+        ];
+
+        mutator.acqUnits.GET({
+          params: {
+            limit: LIMIT_MAX,
+            query: buildAcqUnitsQuery(units, isEdit),
+          },
+        });
+      });
+    }
+  });
+
+  const units = get(resources, 'acqUnits.records', []);
+
+  return (
+    <AcqUnitsField
+      name={name}
+      units={units}
+      disabled={perm ? stripes.hasPerm(perm) : false}
+    />
+  );
+};
+
+AcqUnitsFieldContainer.manifest = Object.freeze({
```

Comment:
I think it's the first time this repo is connected to back-end. Maybe it makes sense to put okapi interface dependency in package.json

Reply:
done